### PR TITLE
Feature: folder section header view

### PR DESCRIPTION
**Pull request by billypchan**
## What's new in this PR?

Create collection view section header for folder feature. The Section header's arrow is points to right when the `collapsed` flag is set to true.

### CODE REVIEW

File: Wire-iOS/Sources/UserInterface/ConversationList/ListContent/ConversationListContentController/Model/ConversationListViewModel.swift [hunk 1]
```diff
@@ -109,6 +109,34 @@ final class ConversationListViewModel: NSObject {
         conversationDirectoryToken = userSession.conversationDirectory.addObserver(self)
     }
 
+    func sectionHeaderTitle(sectionIndex: Int) -> String? {
```
Comment (typfel) [inline Wire-iOS/Sources/UserInterface/ConversationList/ListContent/ConversationListContentController/Model/ConversationListViewModel.swift]: Why not make this a property on the `Section` struct same thing goes for if it has section header or not.

Comment (billypchan) [reply Wire-iOS/Sources/UserInterface/ConversationList/ListContent/ConversationListContentController/Model/ConversationListViewModel.swift]: moved under `Section.Kind.title` property

Comment (typfel) [reply Wire-iOS/Sources/UserInterface/ConversationList/ListContent/ConversationListContentController/Model/ConversationListViewModel.swift]: Not pushed yet?

File: Wire-iOS/Sources/UserInterface/ConversationList/ListContent/ConversationListContentController/Model/ConversationListViewModel.swift [hunk 2]
```diff
@@ -42,6 +42,19 @@ final class ConversationListViewModel: NSObject {
 
             ///TODO:
             //    case customFolder(folder: FolderType)
+            
+            var title: String? {
```
Comment (billypchan) [inline Wire-iOS/Sources/UserInterface/ConversationList/ListContent/ConversationListContentController/Model/ConversationListViewModel.swift]: @typfel the localized title is moved to here.

**Pull request merged**